packages/roosterjs-editor-api/lib/format/setAlignment.ts

```diff
@@ -14,23 +18,77 @@ import {
  * Alignment.Center, Alignment.Left, Alignment.Right
  */
 export default function setAlignment(editor: IEditor, alignment: Alignment) {
-    let command = DocumentCommand.JustifyLeft;
-    let align = 'left';
+    const element = editor.getElementAtCursor();
+    const selection = editor.getSelectionRangeEx();
+    const selectionType = selection.type;
+    editor.addUndoSnapshot(() => {
+        if (
+            editor.isFeatureEnabled(ExperimentalFeatures.TableAlignment) &&
+            selection &&

```

Comment:
this need to be moved before line 23. you already used selection before check it